filter-assingment/Dhanashri/js/index.js

```diff
@@ -0,0 +1,252 @@
+*/
+const singleStoreItem = (productObj) =>{
+    
+    let mainWrapper = document.createElement('div');
+    setAttributes(mainWrapper,{'class':'col-10 col-sm-6 col-lg-4 mx-auto my-3 storeItem'});
+
+    let card = document.createElement('div');
+    setAttributes(card,{'class':'card'});
+
+    let imgContainer = document.createElement('div');
+    setAttributes(imgContainer,{'class':'img-container'});
+
+    let itemImg = document.createElement('img');
+    setAttributes(itemImg,{'src': 'img/' + productObj.image, 'alt': 'Cart Item Image','class': 'card-img-top store-img'});
+
+    let storeIconWrap = document.createElement('span');
+    setAttributes(storeIconWrap,{'class':'store-item-icon'});
+
+    let storeIcon = document.createElement('i');
+    setAttributes(storeIcon,{'class':'fa fa-shopping-cart', 'onclick': 'addIntoCartItem(\'' + productObj.image + '\',\'' + productObj.name + '\',' + productObj.price + ')'});
+
+    let cardBody = document.createElement('div');
+    setAttributes(cardBody,{'class':'card-body'});
+
+    let cardBodyInnerWrap = document.createElement('div');
+    setAttributes(cardBodyInnerWrap,{'class':'card-text d-flex justify-content-between text-capitalize'});
+
+    let itemName = document.createElement('h5');
+    setAttributes(itemName,{'id':'store-item-name'});
+
+    let itemNameValue = document.createTextNode(productObj.name);
+
+    let itemPriceWrap = document.createElement('h5');
+    setAttributes(itemPriceWrap,{'class':'store-item-value'});
+    
+    let priceSymb = document.createTextNode('$');
+
+    let itemPrice = document.createElement('strong');
+    setAttributes(itemPrice,{'class':'font-weight-bold', 'id':'store-item-price'});
+
+    let itemPriceValue = document.createTextNode(productObj.price);
+
+
+    mainWrapper.appendChild(card);
+    card.appendChild(imgContainer);
+    imgContainer.appendChild(itemImg);
+    imgContainer.appendChild(storeIconWrap);
+    storeIconWrap.appendChild(storeIcon);
+    document.body.appendChild(mainWrapper);
+    card.appendChild(cardBody);
+    cardBody.appendChild(cardBodyInnerWrap);
+    cardBodyInnerWrap.appendChild(itemName);
+    itemName.appendChild(itemNameValue);
+    cardBodyInnerWrap.appendChild(itemPriceWrap);
+    itemPriceWrap.appendChild(priceSymb);
+    itemPriceWrap.appendChild(itemPrice);
+    itemPrice.appendChild(itemPriceValue);
+
+    singleItem.appendChild(mainWrapper);  
+}
+
+
+let filterArray = [];
+
+let getFilteredProduct = (cat) => {
+    filterArray = [];
+    if(cat == "all"){
+        let productList = storeItem1.map(function(data,index){
+            data.item.map(function(data1,index1){
+                filterArray.unshift(data1);
+            });
+        });
+    }else{
+        for(let i=0;i<storeItem1.length;i++){
+            if(storeItem1[i].category == cat){
+                let productList = storeItem1[i].item.map(function(data,index){
+                    filterArray.unshift(data);
```

Comment:
I am dicey about unshift and shift for now it is fine but as good practice we should avoid use of it. 

Also why do we need nested loops , I could see for..loop includes map which is another loop.